### DISCUSSION TITLE
How to handle exceptions?

### DISCUSSION THREAD

**Discussion by ansorre** [comfyanonymous/ComfyUI#3643] [Q&A]
Is there a way to catch an exception and resume execution from a specific node? For example, my workflow requires a face restore at some point, and yet a face is not always present in the image. When the image does not contain a face, the entire workflow hangs due to an exception raised by ComfyUI complaining that there is no face.

**Comment by BHMW** [comment]
Did you find any solution for this? running into the same problem

**Reply by ltdrdata** [reply]
What is your workflow?

**Reply by ansorre** [reply]
No way, I didn't, this is still a serious problem in ComfyUI.

**Comment by Debojyoti-Bhattacharyya** [comment]
> Is there a way to catch an exception and resume execution from a specific node? For example, my workflow requires a face restore at some point, and yet a face is not always present in the image. When the image does not contain a face, the entire workflow hangs due to an exception raised by ComfyUI complaining that there is no face.

Can you please add some screenshots?

**Reply by ansorre** [reply]
Here we go. The promt is "a sunny day on the beach in a tropical island" and since there are no faces the exception rises.

![Screenshot_13](https://github.com/user-attachments/assets/b3aa3b23-5704-45c9-afb7-9a04febdaf6e)

**Comment by AngryLoki** [comment]
I guess with https://github.com/comfyanonymous/ComfyUI/issues/4230 it should be possible to write if/else nodes and even handle `for face in faces` loops with nodes. It would be interesting to see some new solutions.
